tests/no-semi/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,583 @@
+x
+;({ a: 1 }.entries())
+x
+;({ a: 1 }.entries())
+x
+;<Hello />
+x
+;\`string\`
+
+// doesn't have to be preceded by a semicolon
+
+class X {}
+;[1, 2, 3].forEach(fn)
+
+// TODO: enable lang features
+// x; ::b.c
+// TODO: upgrade parser
+// class A {
+//   async; // The semicolon is *not* necessary
+//   x(){}
+// }
+// class B {
+//   static; // The semicolon *is* necessary
+//   x(){}
+// }
+
+// don't semicolon if it doesn't start statement
+
+if (true) (() => {})()
+
+class X {
+  a = 0
+  ;[\\"b\\"] = 1
+  ;[\\"m\\"]() {
+    1
+  }
+  ;*g() {}
+}
+
+// check indentation
+
+if (true) {
+  x
+  ;(() => {})()
+}
+
+// check that it doesn't break non-ASI
+
+1 - 1
+
+1 + 1
+
+1 / 1
+
+arr[0]
+
+fn(x)
+
+!1
+
+1 < 1
+
+tag\`string\`
+
+class X {
+  get x() {}
+
+  get [\\"y\\"]() {}
+  set [\\"z\\"](z) {}
+  async [\\"a\\"]() {}
+  async *g() {}
+
+  a = 0
+  b = 1
+}
+
+while (false)
+  (function() {})()
+
+aReallyLongLine012345678901234567890123456789012345678901234567890123456789 *
+  (b + c)
+"
+`;
+
+exports[`no-semi.js 3`] = `
+"
+// with preexisting semi
+
+x; [1, 2, 3].forEach(fn)
+x; /r/i.test('r')
+x; +1
+x; - 1
+x; ('h' + 'i').repeat(10)
+x; (1, 2)
+x; (() => {})()
+x; ({ a: 1 }).entries()
+x; ({ a: 1 }).entries()
+x; <Hello />
+x; \`string\`
+
+// doesn't have to be preceded by a semicolon
+
+class X {} [1, 2, 3].forEach(fn)
+
+// TODO: enable lang features
+// x; ::b.c
+// TODO: upgrade parser
+// class A {
+//   async; // The semicolon is *not* necessary
+//   x(){}
+// }
+// class B {
+//   static; // The semicolon *is* necessary
+//   x(){}
+// }
+
+
+// don't semicolon if it doesn't start statement
+
+if (true) (() => {})()
+
+class X {
+  a = 0;
+  ['b'] = 1;
+  ['m']() { 1 };
+  *g() {}
+}
+
+// check indentation
+
+if (true) {
+  x; (() => {})()
+}
+
+// check that it doesn't break non-ASI
+
+1
+- 1
+
+1
++ 1
+
+1
+/ 1
+
+arr
+[0]
+
+fn
+(x)
+
+!1
+
+1
+< 1
+
+tag
+\`string\`
+
+class X {
+  get
+  x() {}
+
+  get ['y']() {}
+  set ['z'](z) {}
+  async ['a']() {}
+  async *g() {}
+
+  a = 0
+  b = 1
+}
+
+while (false)
+  (function(){}())
+
+aReallyLongLine012345678901234567890123456789012345678901234567890123456789 *
+  (b + c)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// with preexisting semi
+
+x
+;[1, 2, 3].forEach(fn)
+x
+;/r/i.test(\\"r\\")
+x
+;+1
+x
+;-1
+x
+;(\\"h\\" + \\"i\\").repeat(10)
+x
+1, 2
+x
+;(() => {})()
+x
+;({ a: 1 }.entries())
+x
+;({ a: 1 }.entries())
+x
+;<Hello />
+x
+;\`string\`
+
+// doesn't have to be preceded by a semicolon
+
+class X {}
+;[1, 2, 3].forEach(fn)
+
+// TODO: enable lang features
+// x; ::b.c
+// TODO: upgrade parser
+// class A {
+//   async; // The semicolon is *not* necessary
+//   x(){}
+// }
+// class B {
+//   static; // The semicolon *is* necessary
+//   x(){}
+// }
+
+// don't semicolon if it doesn't start statement
+
+if (true) (() => {})()
+
+class X {
+  a = 0
+  ;[\\"b\\"] = 1
+  ;[\\"m\\"]() {
+    1
+  }
+  ;*g() {}
```

Comment:
Hmm, I kind of think it'd be worthwhile to break the "just put a `;` at the start of every line" rule for class methods (especially since the universe of possibilities is far smaller). 

@bakkot @lydell what do you think?